### ISSUE TITLE
switch to pane by name

### ISSUE THREAD

**Issue by davxy** [zellij-org/zellij#2779]
Currently zellij cli allows to switch to prev or next pane.

```
zellij action focus-next-pane
zellij action focus-previous-pane
```

When you have a bunch of panes and you use the command in a script for doing more complex stuff this looses a bit its utility.

Would be nice to be able to do something like:

```
zellij action focus-pane <name>
```

Probably it'd be useful the same for tabs as well

**Comment by Zykino** [comment]
You should check out the session manager that comes as a default plugin in 0.38. You can see a tree of session-tab-pane and fuzzy find whatever you want.

The default keybinding is `Ctrl + o, w`

**Comment by davxy** [comment]
Yeah, I'm using that. But I require a CLI command to be used from a script to select a pane by name

**Comment by Zykino** [comment]
Oh yeah. I don’t think it is currently possible. Especially because zellij is truly multi-user and it is hard to know which user typed the 

I’m developing a zellij-console plugin that I hope will enable better integration even from the cli… but that part is only in project for now.

If you want a specialized plugin for that, you could/should go ahead ! I think we could use the `configuration=foo` to achieve that… but I did not tested yet. I’m sure better API will be created if the need is reported :).

**Comment by davxy** [comment]
> Oh yeah. I don’t think it is currently possible. Especially because zellij is truly multi-user and it is hard to know which user typed the

If the multi-user thing is the problem, isn't there the same issue with `zellij action focus-next-pane`?

> I’m developing a zellij-console plugin that I hope will enable better integration even from the cli… but that part is only in project for now.

I had a quick look to your plugin but at first glance is not clear to me what is the end goal (I never tried to create a zellij plugin)

> If you want a specialized plugin for that, you could/should go ahead ! I think we could use the `configuration=foo` to achieve that… but I did not tested yet. I’m sure better API will be created if the need is reported :).

Yeah probably I'll try to implement something. Do you know if I can have construct a layout with a pane hidden and then I can toggle its visibility with a key binding?

**Comment by Zykino** [comment]
>> Oh yeah. I don’t think it is currently possible. Especially because zellij is truly multi-user and it is hard to know which user typed the

> If the multi-user thing is the problem, isn't there the same issue with `zellij action focus-next-pane`?

It looks like it does not work (well) to me: open `zellij`, in another terminal connect to the same session (`zellij a` or other). If I run the command you gave in the second terminal, the comportment is inconsistent:
Having 2 tabs, the first one have 2 panes and the second have only 1.
1. If both terminal are focused on the exact same pane, both terminal focus the next pane (on tab1 it goes left/right, on tab2 nothing happens). :heavy_check_mark: I guess
2. If Term1 is on T1,P1, and Term2 sends the command from T1,P2, then only Term1 have its focus moving to T1,P2. :thinking: that sounds weird already…
3. If Term1 is on T1,P1, and Term2 sends the command from T2’s only pane, then Term1 change it’s focused pane. :x: that definitely isn’t what anyone would expect.

> Yeah probably I'll try to implement something. Do you know if I can have construct a layout with a pane hidden and then I can toggle its visibility with a key binding?

I’m not 100% sure, but I think you can. I think I heard in a future version it will be possible to have plugins fully hidden.
